### PR TITLE
Update to build cleanly

### LINKED ISSUE #10: #include <ESP8266WebServer.h> can be replaced with #include <WebServer.h>
I just got it running with WebServer.h from espressif for the esp32 without changing anything in the code. Just

#include <ESP8266WebServer.h>
to #include <WebServer.h>

### LINKED ISSUE #20: Lots of warnings when building
There are a number of warnings building this repository.  For example:

### Missing [field initializers](https://en.cppreference.com/w/cpp/language/default_initialization)

In fields.h, while FieldList is an array, the array elements are each a POD struct of type Field, which has no non-default constructor.  Therefore, according to the specification:
> The effects of default initialization are: 
> …
> otherwise, nothing is done: the objects with automatic storage duration (and their subobjects) are initialized to indeterminate values. 

### Comparisons always false due to limited range of data type

Simply put, the `long` result from `string.toInt()` was placed into an `uint8_t`, prior to clamping the value.  This change stores the conversion result in a `long` to perform the clamping prior to storing the value in the `uint8_t` field.

### "warning: extra tokens at end of #include directive"

No semi-colons after #include "file.h"

### "error: multi-line comment [-Werror=comment]"

Don't allow the last character of a comment line to end in a backslash character.

### "warning: 'typedef' was ignored in this declaration"

The `typedef` of `Field` is in the wrong spot.

### PR DESCRIPTION
**Pull request by henrygab**
Fixes #20
Fixes #10

### Use `webserver.h` instead of `ESP8266WebServer.h`

Makes this build without dependencies.

### Fix `Missing field initializers` warnings.

In fields.h, while FieldList is an array, the array elements are each a POD struct of type Field, which has no non-default constructor. Therefore, according to the specification:
> The effects of default initialization are:
> …
> otherwise, nothing is done: the objects with automatic storage duration (and their subobjects) are initialized to indeterminate values.

### Fix `Comparisons always false due to limited range of data type` warning

Simply put, the long result from string.toInt() was placed into an uint8_t, prior to clamping the value. This change stores the conversion result in a long to perform the clamping prior to storing the value in the uint8_t field.

### Fix `warning: extra tokens at end of #include directive`

Remove semi-colons from #include lines

### Fix `error: multi-line comment [-Werror=comment]`

Ensure the last character of a comment line is not a backslash character.

### Fix `warning: 'typedef' was ignored in this declaration`

Move the name `Field` to follow the struct definition.

### CODE REVIEW

File: fields.h [hunk 1]
```diff
@@ -76,9 +76,15 @@ String getPalette() {
...
-  else if (currentPaletteIndex >= paletteCount) currentPaletteIndex = paletteCount - 1;
+  // value.toInt() returns long, while currentPaletteIndex is declared as uint8_t
+  long tmp = value.toInt();
+  // clamp to [0.. paletteCount-1]
+  if (tmp < 1) {
+    tmp = 1;
+  } else if (tmp > (paletteCount+1)) {
+    tmp = paletteCount+1;
+  }
+  currentPaletteIndex = tmp;
```
Comment (jasoncoon) [inline fields.h]: Thanks for this pull request!  Shouldn't this actually clamp to `[0.. paletteCount-1]` (as noted in the comment( instead of `[1.. paletteCount+1]`?

**Review by jasoncoon — CHANGES_REQUESTED** [review]

**Comment by henrygab** [comment]
@jasoncoon Fixed.  Thanks for review and catching that!

**Review by jasoncoon — APPROVED** [review]
Sorry for the massive delay, thank you!
